lua/oil/config.lua

```diff
@@ -100,6 +100,9 @@ local default_config = {
       { "type", "asc" },
       { "name", "asc" },
     },
+    -- Read the first line of the file and use vim's matching ft.
+    -- Can slow down the execution considerably for large projects.
+    use_vim_ft = false,
```

Comment:
Could we instead make this an option for the `icon` column, like we do for `add_padding`? You can just access it on the `conf` table like `conf.use_slow_filetype_detection` and then add an entry to [this file](https://github.com/stevearc/oil.nvim/blob/685cdb4ffa74473d75a1b97451f8654ceeab0f4a/scripts/generate.py#L140-L152) to generate the documentation for it.